### ISSUE TITLE
[ED2] ggplot2 or base R figures for Ed2?

### ISSUE THREAD

**Issue by mine-cetinkaya-rundel** [OpenIntroStat/randomization-and-simulation#3]
@beanumber @andrewpbray @hardin47

What are people's thoughts on converting all figures to ggplot2 in the book? From an efficiency perspective I vote no -- it's not trivial to convert all existing figures while keeping a unified look throughout the book. I can guarantee you with all the customization needed it will end up being very time consuming, in some cases only to achieve basically the same outcome as before.

However this will mean the DataCamp course figures will look different. To me this is ok since while they stylistically may look different, the story they're telling will be the same. However I wanted to raise this as an issue and make a firm decision before going forward.

Basically the code needed to generate the multivariate figures in base R is not something I would dare teach students learning from this book, but I am ok with hiding that behind the images in the book and reveal ggplot2 code in things like DataCamp and labs.

**Comment by beanumber** [comment]
If it will be time consuming then I vote "no". 

In principle, I would favor creating the plots in `ggplot2` and then writing a custom `theme_openintro()` function.

**Comment by mine-cetinkaya-rundel** [comment]
The time consuming bit is customizing how a plot falls on a page - some of which could be addressed with a theme, some of which would probably require additional tweaking. There are many custom tweaks in the plots in the book in order to save a bit of space -> page -> sales price.

And changing the plots would probably mean we would need to update pretty much every single page, so that will add to time.

I agree in principle it would be better, though if from student learning perspective it won't actually have an impact I feel like I'd rather put in the time elsewhere.

**Comment by hardin47** [comment]
I'm not sure if this is a wrench or not, but…  Isn't it true that if we change the order around / add a few things here and there that almost every single page will change anyway?

--
Jo Hardin
Professor, Department of Mathematics
Pomona College
610 N. College Ave.
Claremont, CA 91711
909-607-8717
jo.hardin@pomona.edu<mailto:jo.hardin@pomona.edu>
http://research.pomona.edu/johardin<http://pages.pomona.edu/~jsh04747>

**Comment by mine-cetinkaya-rundel** [comment]
@hardin47 the page number will change (which is no problem), but the content doesn't need to vs. if we want to update every single plot we will need to go update the code for each one and check how the new plot looks on the page, and then potentially resize. the fact that this might move the page number of a piece of content isn't a huge issue however it is a time consuming process.

**Comment by beanumber** [comment]
It would be less time-consuming to resize the plots if they were created in a chunk within a single `.Rnw` file for each chapter. But here again, to create those files would be a lot of work. So I'm with @mine-cetinkaya-rundel on this one -- I say we push to the third edition.
